src/number_theory/cyclotomic/rat.lean

```diff
@@ -135,11 +136,12 @@ local attribute [instance] algebra_rat_subsingleton
 lemma cyclotomic_ring_is_integral_closure_of_prime_pow :
```

Comment:
This must be related to what happened to `char_zero`. In the current `master` Lean is able to find `algebra ℚ(cyclotomic_field (p ^ k) ℚ)` here, even if there is
```
local attribute [-instance] cyclotomic_field.algebra
```
First of all this doesn't work anymore (I don't know exactly why). Indeed I don't understand who Lean knows it, I thought it goes through `[algebra_rat](https://leanprover-community.github.io/mathlib_docs/algebra/algebra/basic.html#algebra_rat)`, but then I don't know how it finds the `char_zero` instance. The usual trick of doing `letI` and having a look at the infoview doesn't work for me, since it finds `cyclotomic_field.algebra` (that we don't want because of `char_zero` diamond) or it shows a circular arguments (via `char_zero` and `number_field`) that of course it's the one it really uses.

Reply:
Ah, I guess it knows that it is a `number_field`, so it has `char_zero` and so it is a `ℚ`-algebra. Interestingly, it can forget about the `ℚ`-algebra structure given by `cyclotomic_field.algebra` but it doesn't forget the `number_field` instance, since the statement doesn't require a `ℚ`-algebra.